langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java

```diff
@@ -97,6 +115,15 @@ public static BArray findAllGroups(BRegexpValue regExp, BString str, long startI
         return groupArray;
     }
 
+    private static Matcher getMatcher(BRegexpValue regExp, BString str) {
```

Comment:
Why do we have this here? Can't we use RegexpUtil's version

Reply:
Was handled in #39444. 
Re-synced from the master and verified that this is fixe.